### PR TITLE
Collection::every() should return false for empty collection.

### PR DESCRIPTION
**Pull request by ADmad**
Refs https://github.com/cakephp/cakephp/issues/8632#issuecomment-210149982

### CODE REVIEW

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -86,7 +86,11 @@ public function reject(callable $c)
      */
     public function every(callable $c)
     {
-        foreach ($this->unwrap() as $key => $value) {
+        $iterator = $this->unwrap();
+        if (count($iterator) === 0) {
```
Comment (lorenzo) [inline src/Collection/CollectionTrait.php]: iterators are not countable by default

File: src/Collection/CollectionTrait.php [hunk 2]
```diff
@@ -86,7 +86,11 @@ public function reject(callable $c)
...
+            return false;
+        }
+        foreach ($iterator as $key => $value) {
```
Comment (lorenzo) [inline src/Collection/CollectionTrait.php]: better set a boolean here as the first line of this loop. If the boolean is still false at the end of the loop, that means the iterator was empty

File: src/Collection/CollectionTrait.php (hunk 1 above)
Comment (ADmad) [inline src/Collection/CollectionTrait.php]: :unamused:

**Comment by ADmad** [comment]
@lorenzo Updated

**Comment by markstory** [comment]
Looks good to me. :+1:

**Pull request merged**